src/components/base-table/BaseTable.js

```diff
@@ -0,0 +1,97 @@
+/* @flow */
+import * as React from 'react';
+import classNames from 'classnames';
+import Table from 'react-virtualized/dist/es/Table';
+import pure from '../pure-component';
+import 'react-virtualized/styles.css';
+import './Table.scss';
+
+type Props = {
+    autoHeight?: boolean,
+    children?: any, // required
+    className?: string,
+    disableHeader?: boolean,
+    headerClassName?: string,
+    headerHeight?: number,
+    height: number,
+    isStriped: boolean,
+    onRowClick?: Function,
+    onRowsRendered?: Function,
+    onScroll?: Function,
+    rowCount: number,
+    rowGetter: Function,
+    rowHeight?: number,
+    rowRenderer?: Function,
+    scrollTop?: number,
+    sortColumn?: string,
+    sortDirection?: string,
+    width: number,
+};
+
+const BaseTable = pure(
+    ({
+        autoHeight = false,
+        children,
+        className = '',
+        disableHeader = false,
+        headerClassName = '',
+        headerHeight = 40,
+        height,
+        isStriped = false,
+        onRowClick,
+        onRowsRendered,
+        onScroll,
+        rowCount,
+        rowGetter,
+        rowHeight = 50,
+        rowRenderer,
+        scrollTop,
+        sortColumn,
+        sortDirection = 'ASC',
+        width,
+    }: Props) => {
+        const rowClassName = ({ index }) => {
+            if (index === -1) {
+                return 'table-header';
+            }
+            let rowClass = 'table-row';
+            if (isStriped && index % 2 === 0) {
+                // can't target odd elements in css because row elements can disappear
+                rowClass += ' has-stripe';
+            }
+            return rowClass;
+        };
+
+        return (
+            <Table
```

Comment:
Would a Grid be better than a Table?
https://bvaughn.github.io/react-virtualized/#/components/Grid

Since we are implementing a grid view

Reply:
Actually now they have this layout too https://bvaughn.github.io/react-virtualized/#/components/Masonry
Pinterest style, maybe we can use this but with fixed cards. It also seems responsive, so you may not need all the media scss stuff

Reply:
Here is a striped down/hacked version from their examples: https://codesandbox.io/s/distracted-breeze-o8lxy

Reply:
It seems that both Collection and Masonry are better suited for variable sized cells, and don't have the concept of rows.  Reading further, Table actually imports and uses Grid internally.  Grid is designed to be scrolled both vertically and horizontally.  Table is more optimized than Grid, but is not designed to scroll horizontally.  The GridView for Content Explorer shouldn't need to scroll horizontally, so Table seems to be the best option.

Reply:
But table has fixed columns and headers. Whereas a grid view has no concept for that and the columns should flow to the next row when viewport size changes. Our cells will also be variable sized depending upon if it's a file, folder or weblink

Reply:
BTW if you see that example link above, it only scrolls vertically and moves cells to a new row if viewport reduces